modules/oe_theme_content_event/src/Plugin/ExtraField/Display/DetailsExtraField.php

```diff
@@ -119,6 +119,11 @@ protected function addRenderableLocation(array &$build, ContentEntityInterface $
       return;
     }
 
+    // Inherit language from parent entity.
+    if ($venue->hasTranslation($entity->language()->getId())) {
+      $venue = $venue->getTranslation($entity->language()->getId());
+    }
+
```

Comment:
Instead of this inject the entity repository service and do
```php
  protected function addRenderableLocation(array &$build, ContentEntityInterface $entity): void {
    if ($entity->get('oe_event_venue')->isEmpty()) {
      return;
    }

    $venue = $entity->get('oe_event_venue')->entity;
    if (!$venue instanceof VenueInterface) {
      return;
    }

    $venue = $this->entityRepository->getTranslationFromContext($venue);
    $cacheability = CacheableMetadata::createFromRenderArray($build);
```

Reply:
done